src/python/pants/backend/scala/dependency_inference/ScalaParser.scala

```diff
@@ -248,7 +248,7 @@ class SourceAnalysisTraverser extends Traverser {
         recordProvidedName(name)
       })
       decltpe.foreach(tpe => {
-        recordConsumedSymbol(extractName(tpe))
+        extractNamesFromTypeTree(tpe).foreach(recordConsumedSymbol(_))
```

Comment:
Maybe add a test for this type of construct to the existing parser test?

Reply:
I'm working on it. FYI, the `Defn.Var` few lines below is using `extractNamesFromTypeTree` as well.

Reply:
done